backend/src/middleware/checkToken.ts

```diff
@@ -0,0 +1,24 @@
+import { Request, Response } from 'express';
+import { readToken } from './../services/tokenService';
+
+/**
+ * This middleware prevents anonymous users
+ * from accessing sensitive API endpoints.
+ * The sensitive endpoints must be added to the app
+ * after this middleware.
+ * Endpoints that must be available for anonymous users
+ * must be added before this middleware.
+ */
+const checkToken = async (
+    req: Request,
+    res: Response,
+    next: (param?: unknown) => void
+): Promise<void> => {
+    //we want this to stop requests only to api endpoints
+    //and not to express.static for example
+    if (req.path.startsWith('/api/') && !(await readToken(req)))
```

Comment:
Can you not exclude the API's you don't want to be token checked instead of checking pathname, by putting them before the middleware or some other way? This check is a code smell as if we want to protect something else with the middleware too other than /api/, it will mysteriously not work because the limitation is not even documented.